vmm/src/vm.rs

```diff
@@ -548,6 +557,17 @@ impl Vm {
             )
             .map_err(Error::CpuManager)?;
 
+        let load_payload_handle = if snapshot.is_none() {
+            Self::load_payload_async(
+                &memory_manager,
+                &config,
+                #[cfg(feature = "igvm")]
+                &cpu_manager,
+            )?
+        } else {
+            None
+        };
+
```

Comment:
No sure this is right to push `load_payload_async()` in this way and making it dependent on `CpuManager`. What do you need from CpuManager for loading `igvm` file?

Reply:
We need to call get_cpuid_leaf from laod_igvm function. cpu_manager has that function implemented.

Reply:
We want to have start loading as early as possible for better boot time. 

With a quick glance, looks like you don't need per-vcpu specific cpuid information, right? You can generate the common cpuid via `arch::generate_common_cpuid()` and pass it down to your `igvm_loader` module for retrieving the cpuid info.

Reply:
MSHV does not provide get_supported_cpuid yet. Don't you think that will not serve the purpose here?

Reply:
hmm... How hard is it for MSHV to add such support? Basically, we are asking the ability to query supported CPUIDs from the hypervisor before creating vCPU.

Reply:
This support needs to be added into Kernel Driver. We can create an issue for now and start working on this. Is it okay @likebreath ?

Reply:
I think the effect on the performance is going to be negligible. Creating the CPU manager takes a few milliseconds at best, and there is a lot of other things the main thread needs to do.

Reply:
Once we get the support from the hypervisor to query supported CPUIDs, we can unify the code path with KVM.

Reply:
Okay. I am fine with refactoring it later when the capability is added from `mshv`.

For now, I'd prefer to keep the current code path unchanged (mostly for kvm) when `igvm` is not enabled. We can differientiate two cases with feature gate `#[cfg(not(feature = "igvm"))]` and `#[cfg(feature = "igvm")]`.

Reply:
Do you suggest to keep all the related functions separate?

Reply:
I am talking about not delaying `load_payload_async()` when `igvm` is not enabled. The way how `load_payload_async()` itself is augmented with `igvm` feature gate is perfectly fine.